components/director/internal/domain/tenant/fetcher_on_demand.go

```diff
@@ -0,0 +1,43 @@
+package tenant
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/pkg/errors"
+)
+
+// Client is responsible for making HTTP requests.
+//go:generate mockery --name=Client --output=automock --outpkg=automock --case=underscore
+type Client interface {
+	Do(req *http.Request) (*http.Response, error)
+}
+
+type fetchOnDemandService struct {
+	client           Client
+	tenantFetcherURL string
+}
+
+// NewFetchOnDemandService returns object responsible for fetching tenants
+func NewFetchOnDemandService(client Client, url string) *fetchOnDemandService {
+	return &fetchOnDemandService{
+		client:           client,
+		tenantFetcherURL: url,
+	}
+}
+
+// FetchOnDemand calls an API which fetch details for the given tenant and store them in the database
```

Comment:
Since we don't process the response body, we can rephrase this like:
> FetchOnDemand calls an API which fetches details for the given tenant from an external tenancy service, stores the tenant in the Compass DB and returns 200 OK if the tenant was successfully created.